### ISSUE TITLE
Errors are not detected across modules?

### ISSUE THREAD

**Issue by Anrock** [haskell/haskell-ide-engine#614]
Title may be a little unclear, see repro.

HIE Version 0.2.0.0, Git revision e1b6786a73b1235a05d4d19e888dc982d033c283 (1438 commits) x86_64 ghc-8.2.2

1. Create simple stack project (`stack new test simple`)
2. Add a library section with atleast one module (`Common` in this case) to test.cabal and make executable depend on it:
```cabal
executable test
  hs-source-dirs:      src
  main-is:             Main.hs
  default-language:    Haskell2010
  build-depends:       base >= 4.7 && < 5,
                       test

library
  hs-source-dirs: lib
  default-language: Haskell2010
  build-depends: base
  exposed-modules: Common
```
3. In library module define some type (`Foo`) and a function (`func`) that will break when type is modified:
```haskell
module Common where

data Foo = Bar | Baz deriving Show

func :: Foo -> Foo
func Bar = Baz
func Baz = Bar
```
4. In `Main.hs` import library module and write similar function (`func'`):
```haskell
module Main where

import Common

main :: IO ()
main = putStrLn "hello world"

func' :: Foo -> Foo
func' Bar = Baz
func' Baz = Bar
```
5. Change `Foo` type (`Bar` -> `Bak`) in library module:
```haskell
data Foo = Bak | Baz deriving Show
```

**Expected**: errors reported for both func and func'
**Actual**: error reported only for func in library module
Changing contents of `Main.hs` or relaunching HIE doesn't trigger error, too.
[hie.log](https://bpaste.net/show/6627c0604de8)

I believe this is a regression or at least it's reproducible only in some cases. I've noticed this behaviour only recently - perhaps after exception handling PRs.

**Comment by wz1000** [comment]
Does this only happen when the files are in differing cabal components? Then it might be a limitation of the mapped files hack in HIE/ghc-mod.

**Comment by Anrock** [comment]
@wz1000 no, looks like this happens even if modules are in same component.
I've added `Another` module to library:
```cabal
library
  hs-source-dirs: lib
  default-language: Haskell2010
  build-depends: base
  exposed-modules: Common
  other-modules: Another
```
with contents:
```haskell
module Another where
import Common

func'' :: Foo -> Foo
func'' Bar = Baz
func'' Baz = Bar
```

Changed `Bar` to `Bak` once again and nothing is reported for Another.hs. Here is [log](https://bpaste.net/show/3ea7a46ea79f)
